### PR TITLE
Wait for editor process to finish

### LINKED ISSUE #12: Find out if it worked
If I have a typo (e.g. `export EDITOR=cude`) or I supply some spec that fails horribly it would be nice to know if it worked. E.g. 

```javascript
if (openEditor(['unicorn.js:5:3'])) {
  console.log('Editor opening working');
} else {
  console.warn('Editor opening failed');
}
```
Even better if it could be `try { } catch ` instead of a boolean return.

### PR DESCRIPTION
**Pull request by tandibar**
Add option `wait` so we know when the edit process is finished.

I hope you are fine with this feature. If not, just tell me what I can do better.

Thank you for providing great open source software!

This builds up on my previous PR: https://github.com/sindresorhus/open-editor/pull/16

Fixes #12

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -38,6 +38,10 @@ openEditor([
 openEditor([
 	'unicorn.js:5:3',
 ]);
+
+openEditor([
+	'unicorn.js:5:3',
+], { wait: true }).then(() => console.log("file was closed"));
```
Comment (sindresorhus) [inline readme.md:44]: Use async/await and single-quotes.

Comment (tandibar) [reply readme.md]: of course, see e0a0654

**Comment by sindresorhus** [comment]
You also need to update index.d.ts

**Comment by sindresorhus** [comment]
CI is failing. Make sure you always run `npm test` locally.

**Comment by tandibar** [comment]
Local run throws no errors:
image

**Comment by sindresorhus** [comment]
> Local run throws no errors:

That commits fails for me too locally. You probably need to `rm -rf node_modules` and reinstall.

File: index.d.ts [hunk 2]
```diff
@@ -9,6 +9,13 @@ export interface Options {
 	__Only use this option if you really have to.__ Can be useful if you want to force a specific editor or implement your own auto-detection.
 	*/
 	readonly editor?: string;
+
+	/**
+	Wheather you want to wait for the editor to quit.
+
+	Default: false
```
Comment (sindresorhus) [inline index.d.ts:16]:
```suggestion
	@default false
```

File: index.d.ts [hunk 3]
```diff
@@ -9,6 +9,13 @@ export interface Options {
 	__Only use this option if you really have to.__ Can be useful if you want to force a specific editor or implement your own auto-detection.
 	*/
 	readonly editor?: string;
+
+	/**
+	Wheather you want to wait for the editor to quit.
```
Comment (sindresorhus) [inline index.d.ts:14]: Typo

File: index.d.ts [hunk 4]
```diff
@@ -50,7 +59,7 @@ openEditor([
 ]);
 ```
 */
-export default function openEditor(files: readonly PathLike[], options?: Options): void;
+export default function openEditor(files: readonly PathLike[], options?: Options): void | Promise<void>;
```
Comment (sindresorhus) [inline index.d.ts:62]: Returning a conditional Promise is weird. Let's just make the JS code also always be a promise.

File: index.d.ts [hunk 5]
```diff
@@ -9,6 +9,13 @@ export interface Options {
 	__Only use this option if you really have to.__ Can be useful if you want to force a specific editor or implement your own auto-detection.
 	*/
 	readonly editor?: string;
+
+	/**
```
Comment (sindresorhus) [inline index.d.ts:13]: Readme and index.d.ts docs should be in sync.

File: index.d.ts [hunk 6]
```diff
@@ -33,6 +40,8 @@ Open the given files in the user's editor at specific line and column if support
 
 @param files - Items should be in the format `foo.js:1:5` or `{file: 'foo.js', line: 1: column: 5}`.
 
+@returns void or Promise<void> - If options.wait is true, a promise is returned which resolves as soon as the editor closes.
```
Comment (sindresorhus) [inline index.d.ts:43]: `void or Promise<void>` is not valid TSDoc notation.

**Review by sindresorhus — CHANGES_REQUESTED** [review]

File: index.d.ts (hunk 4 above)
Comment (tandibar) [reply index.d.ts:62]: So what is your suggesstion? To just always return a Promise, even without `wait=true`? So something like 

    return Promise.resolve()

File: index.d.ts (hunk 5 above)
Comment (tandibar) [reply index.d.ts:13]: I am sorry, but I do not see my mistake here. What have I missed?

Comment (sindresorhus) [reply index.d.ts:13]: If you compare the two side-by-side, it's pretty clear. The description is different and the TS docs are missing the code example.

File: index.d.ts (hunk 4 above)
Comment (sindresorhus) [reply index.d.ts:62]: No, make the exported function an async function.

File: readme.md [hunk 7]
```diff
@@ -44,7 +44,7 @@ openEditor([
 
 ### openEditor(files, options?)
 
-Open the given files in the user's editor at specific line and column if supported by the editor. It does not wait for the editor to start or quit.
+Open the given files in the user's editor at specific line and column if supported by the editor. It does not wait for the editor to start or quit unless you specify `wait: true` in the options.
```
Comment (sindresorhus) [inline readme.md:47]: This is also missing from the TS docs.

File: readme.md [hunk 8]
```diff
@@ -56,6 +56,21 @@ Items should be in the format `foo.js:1:5` or `{file: 'foo.js', line: 1: column:
 
 Type: `object`
 
+##### wait
+
+Type: `boolean`\
+Default: false
```
Comment (sindresorhus) [inline readme.md:62]:
```suggestion
Default: `false`
```

**Comment by tandibar** [comment]
Is there anything I can do to get this merged? I think I updated everything according to your suggestions.

**Comment by sindresorhus** [comment]
There are still unresolved stuff:

- https://github.com/sindresorhus/open-editor/pull/17/files#r866537663 The return type should be changed to a promise.

File: index.d.ts [hunk 9]
```diff
@@ -9,6 +9,13 @@ export interface Options {
 	__Only use this option if you really have to.__ Can be useful if you want to force a specific editor or implement your own auto-detection.
 	*/
 	readonly editor?: string;
+
+	/**
+	Whether you want to wait for the editor to quit.
```
Comment (sindresorhus) [inline index.d.ts:14]: This is not in sync with the readme.

---


> Readme and index.d.ts docs should be in sync. - https://github.com/sindresorhus/open-editor/pull/17#discussion_r866538493

File: index.d.ts [hunk 10]
```diff
@@ -29,10 +36,14 @@ export interface EditorInfo {
 }
 
 /**
-Open the given files in the user's editor at specific line and column if supported by the editor. It does not wait for the editor to start or quit.
+Open the given files in the user's editor at specific line and column if supported by the editor. It does not wait for the editor to start or quit  unless you specify `wait: true` in the options.
```
Comment (sindresorhus) [inline index.d.ts:39]:
```suggestion
Open the given files in the user's editor at specific line and column if supported by the editor. It does not wait for the editor to start or quit unless you specify `wait: true` in the options.
```

File: index.d.ts [hunk 11]
```diff
@@ -29,10 +36,14 @@ export interface EditorInfo {
...
 
 @param files - Items should be in the format `foo.js:1:5` or `{file: 'foo.js', line: 1: column: 5}`.
 
+@param {boolean} [options.wait=false] - Wait until the editor is closed (default: false).
+
+@returns Promise<void> - If options.wait is true, a promise is returned which resolves as soon as the editor closes.
```
Comment (sindresorhus) [inline index.d.ts:45]: The JS code always returns a promise, so this is incorrect.

File: index.d.ts [hunk 12]
```diff
@@ -48,9 +59,13 @@ openEditor([
 openEditor([
 	'unicorn.js:5:3',
 ]);
+
+await openEditor(['unicorn.js:5:3'], {wait: true});
+console.log('File was closed');
+
```
Comment (sindresorhus) [inline index.d.ts:65]:
```suggestion
```

File: index.d.ts [hunk 13]
```diff
@@ -29,10 +36,14 @@ export interface EditorInfo {
 }
 
 /**
-Open the given files in the user's editor at specific line and column if supported by the editor. It does not wait for the editor to start or quit.
+Open the given files in the user's editor at specific line and column if supported by the editor. It does not wait for the editor to start or quit  unless you specify `wait: true` in the options.
 
 @param files - Items should be in the format `foo.js:1:5` or `{file: 'foo.js', line: 1: column: 5}`.
 
+@param {boolean} [options.wait=false] - Wait until the editor is closed (default: false).
+
```
Comment (sindresorhus) [inline index.d.ts:44]:
```suggestion
```

This is moot. It's already documented in the options type.

File: index.js [hunk 14]
```diff
@@ -21,23 +21,46 @@ export function getEditorInfo(files, options = {}) {
...
+			editorArguments.push(
+				'--line',
+				stringifyLineColumnPath(parsed, {
+					file: false,
+				}),
+				parsed.file,
+			);
+
+			if (options.wait) {
+				editorArguments.push('-w');
```
Comment (sindresorhus) [inline index.js:52]: Use the long-flag.

**Review by sindresorhus — CHANGES_REQUESTED** [review]

File: index.js (hunk 14 above)
Comment (tandibar) [reply index.js]: done

File: index.d.ts (hunk 11 above)
Comment (tandibar) [reply index.d.ts]: New wording: https://github.com/sindresorhus/open-editor/pull/17/files#diff-7aa4473ede4abd9ec099e87fec67fd57afafaf39e05d493ab4533acc38547eb8R43

**Comment by tandibar** [comment]
OK, I added all your change requests.

File: index.d.ts [hunk 15]
```diff
@@ -48,9 +57,12 @@ openEditor([
 openEditor([
 	'unicorn.js:5:3',
 ]);
+
+await openEditor(['unicorn.js:5:3'], {wait: true});
+console.log('File was closed');
```
Comment (sindresorhus) [inline index.d.ts:62]: This should instead be an example in the docs for the `wait` option, like in the readme.

**Comment by sindresorhus** [comment]
You did the opposite of what I asked in https://github.com/sindresorhus/open-editor/pull/17#discussion_r1027120894

**Comment by tandibar** [comment]
I hope I did get you now.

**Pull request merged**